docs/python-version-support-policy.rst

```diff
@@ -0,0 +1,88 @@
+`every year in October.`__ New Python versions bring new language and standard
+library features, but they also often remove things the Python development team
+considerd old/deprecated/uncool, and some libraries with C dependencies often
+cause issues (they fail to build or they lack binary wheels on PyPI) for
+shortly after a new Python version. The Python team supports a release for 5
+years after the initial release (3.x.0).
+
+__ https://peps.python.org/pep-0602/
+
+Nikola is used by both Python developers who are using the latest-and-greatest
+Python release, and more casual Python users that use whatever release is
+provided by their Unix-like operating system of choice. Some of them may be
+using rolling-release or often-updated Linux distributions, like Arch Linux or
+Fedora; others might be on Ubuntu LTS or Debian stable.
+
+In order to support the largest group of users, while also keeping the list
+of supported Python releases manageable, and minimizing the number of cases
+where legacy or worse code must be used to support old versions, the following
+Python version support policy is adopted.
+
+Policy
+======
+
+1. Nikola *must* support all Python versions for which at least one of those criteria is true:
+
+   a. It is `supported by the Python core team`__ (5 years since release).
+   b. It is `included`__ in the oldest `Ubuntu LTS release`__  still within standard support (LTS releases are supported for 5 years since release).
+   c. It is included in `Debian stable`__.
+   d. It is included in `Debian oldstable`__, if the that Debian release became oldstable in the past 3 months.
+
+2. Nikola *should* support all Python versions for which at least one of those criteria is true:
+
+   a. It is `the newest Python release supported by PyPy`__.
+   b. It is included in Debian oldstable, if that Debian release has been oldstable since more than 3 months.
+
+3. If a Python version is covered only by the optional criteria, the Nikola maintainers may decide to drop support for it if it is not feasible to maintain.
+
+4. Nikola should also support Python versions that are intermediate versions between the oldest supported version and the latest version.
+   For example, if the latest Python version is 3.9, the oldest officially supported version is 3.7, the but the oldest Ubuntu LTS and Debian stable both have 3.4, then Python 3.5 and 3.6 should *also* be supported.
+
+5. CI pipelines on Linux should be run for all supported Python versions. CI pipelines for Windows and macOS should be run for the latest Python version.
+
+6. In the event Nikola does not support the newest Python release, the relevant issues should be resolved or reported upstream as soon as possible, in consideration of rolling-release distributions that package Nikola.
+
+7. If all mandatory criteria stop appplying to a Python version, it may be dropped after a grace period of 3 months. (This means that Debian oldstable is guaranteed support for a total of 6 months.)
```

Comment:
```suggestion
7. If all mandatory criteria stop applying to a Python version, it may be dropped after a grace period of 3 months. (This means that Debian oldstable is guaranteed support for a total of 6 months.)
```